### PR TITLE
Create Lambda Script + Dashboard

### PR DESCRIPTION
**Pull request by TotalDwarf03**
Create check functions for the dashboard as outlined in github_audit_dashboard.md

### CODE REVIEW

**Comment by TotalDwarf03** [comment]
This PR will also contain changes in Screen Designs as they affect what the policy check function return

File: src/app.py [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+
+import json
+
+import boto3
+from botocore.exceptions import ClientError
+
+st.set_page_config(page_title="GitHub Audit Dashboard", page_icon="./src/branding/ONS-symbol_digital.svg", layout="wide")
+st.logo("./src/branding/ONS_Logo_Digital_Colour_Landscape_Bilingual_RGB.svg")
+
+def get_s3_client() -> boto3.client:
```
Comment (TotalDwarf03) [inline src/app.py:17]: This should be cached

File: aws_lambda/policy_checks.py [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+    """
+    Checks if there are any pull requests in the repository from non-members of the organisation.
+    Returns True if there is an external pull request
+
+    Args:
+        repo_api_url (str): The API endpoint to get the repository's pull requests
+        repo_full_name (str): The fullname of the repository from the GitHub API (format: <owner>/<repo>)
+        gh (api_controller): An instance of the api_controller class to make calls to the GitHub API
+    Returns:
+        bool (True if check breaks policy)
```
Comment (gibbardsteve) [inline aws_lambda/policy_checks.py:280]: Is this actually breaking policy or are we just flagging repos where externals have made PR requests?

**Review by gibbardsteve — COMMENTED** [review]
A few areas for tidy up and a suggestion for changing the secret / dependabot functions to a generic function to common up some of the processing.

File: aws_lambda/README.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+This file needs to be uploaded to AWS ([documentation](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/managing-private-keys-for-github-apps)).
+
+If you do not have access to organisation settings, you need to request a .pem file for the app.
+
+### AWS
+The Lambda Script requires access to an associated Github App secret, this secret is created when the Github App is installed in the appropriate Github Organisation. The contents of the generated pem file is stored in the AWS Secret Manager and retrieved by this service to interact with Github securely.
+
+AWS Secret Manager must be set up with a secret:
+
+/sdp/tools/repoarchive/repo-archive-github.pem
```
[the remark this answers is not in the document]
Comment (TotalDwarf03) [reply aws_lambda/README.md:43]: It does at the moment. It needs some more thought as to how we're going to split up the GitHub apps. Whether we have 1 app to control all 3 GitHub toolings or one per service

File: aws_lambda/policy_checks.py (hunk 2 above)
Comment (TotalDwarf03) [reply aws_lambda/policy_checks.py:280]: This was a requested feature by security. It's also suggested by clause 5.5 in the GitHub Usage Policy

**Review by gibbardsteve — COMMENTED** [review]
Minor typos

**Review by gibbardsteve — APPROVED** [review]
Looks good

**Pull request merged**